src/modm/driver/pressure/ms5837_impl.hpp

```diff
@@ -0,0 +1,154 @@
+    // data.prom.data[7] = modm::fromBigEndian(RF_CALL(readProm(7)));
+
+    MODM_LOG_DEBUG.printf("MS5837 Reset Initialize done\n");
+
+    RF_END_RETURN(data.prom.calculateCrc());
+}
+
+// -----------------------------------------------------------------------------
+
+template < typename I2cMaster >
+modm::ResumableResult<void>
+Ms5837<I2cMaster>::readout(OversamplingRatio osrPressure, OversamplingRatio osrTemperature)
+{
+    RF_BEGIN();
+
+    // start a pressure conversion
+    buffer[0] = i(Command::Convert) | i(Conversion::Pressure) | i(osrPressure);
+    this->transaction.configureWrite(buffer, 1);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
+    }
+
+    // Wait until pressure conversion has finished
+    timeout.restart(std::chrono::milliseconds(conversionDelay[i(osrPressure) >> 1]));
+    RF_WAIT_UNTIL(timeout.isExpired());
+
+    // Get the pressure conversion result from sensor
+    buffer[0] = i(Command::AdcRead);
+
+    this->transaction.configureWriteRead(buffer, 1, data.raw, 3);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
+    }
+
+    // Notify data class about changed buffer
+    data.rawPressureTouched();
+
+
+
+    // start a temperature conversion
+    buffer[0] = i(Command::Convert) | i(Conversion::Temperature) | i(osrTemperature);
+    this->transaction.configureWrite(buffer, 1);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Temperature Conversion failed\n");
+        RF_RETURN();
+    }
+
+
+    // Wait until temperature conversion has finished
+    timeout.restart(std::chrono::milliseconds(conversionDelay[i(osrTemperature) >> 1]));
+    RF_WAIT_UNTIL(timeout.isExpired());
+
+    // Get the temperature conversion result from sensor
+    buffer[0] = i(Command::AdcRead);
+
+    this->transaction.configureWriteRead(buffer, 1, data.raw + 3, 3);
+    if (!RF_CALL( this->runTransaction() )){
+        MODM_LOG_DEBUG.printf("MS5837 Pressure Conversion failed\n");
+        RF_RETURN();
+    }
+
+    // Notify data class about changed buffer
+    data.rawTemperatureTouched();
+
+    RF_END();
+}
+
+// -----------------------------------------------------------------------------
+
+template < typename I2cMaster >
+modm::ResumableResult<uint16_t>
+Ms5837<I2cMaster>::readProm(uint8_t address)
+{
+    RF_BEGIN();
+
+    MODM_LOG_DEBUG.printf("MS5837 readProm(%02x)\n", address);
+
+    buffer[0] = i(Command::PromRead) | ((address & 0b111) << 1);
+    this->transaction.configureWriteRead(buffer, 1, buffer, 2);
+    if (RF_CALL( this->runTransaction() ))
+    {
+        MODM_LOG_DEBUG.printf("MS5837 readProm(%02x) Success: %d (%04x)\n", address, (buffer[1] << 8) | buffer[0], (buffer[1] << 8) | buffer[0]);
+    }
+    else {
+        MODM_LOG_DEBUG.printf("MS5837 readProm(%02x) Failed\n", address);
```

Comment:
Yeah, these seem more useful. Perhaps comment them out, but leave them in, for others to debug, and instead return `0` from here as an error code?